bftengine/src/bftengine/ControllerWithSimpleHistory.cpp

```diff
@@ -226,14 +226,20 @@ bool ControllerWithSimpleHistory::onEndOfEvaluationPeriod() {
   return currentFirstPathChanged;
 }
 
+bool ControllerWithSimpleHistory::insideActiveWindow(const SeqNum& n) { return recentActivity.insideActiveWindow(n); }
+
 void ControllerWithSimpleHistory::onSendingPrePrepare(SeqNum n, CommitPath commitPath) {
+  onSendingPrePrepare(n, commitPath, getMonotonicTime());
+}
+
+void ControllerWithSimpleHistory::onSendingPrePrepare(SeqNum n, CommitPath commitPath, const Time& timePoint) {
   if (!isPrimary || !recentActivity.insideActiveWindow(n)) return;
 
   SeqNoInfo& s = recentActivity.get(n);
 
   // if this is the time we send the preprepare
-  if (s.prePrepareTime == MinTime) {
-    s.prePrepareTime = getMonotonicTime();
+  if (s.prePrepareTime == MinTime && timePoint > MinTime) {
```

Comment:
Two cases I can think of are:

-  Type `Time` is typedef for `std::chrono::steady_clock::time_point` which is capable of representing negative values. one can set the PrePrepeare time to be less than the MinTime.

- MinTime is currently a constant set to 0 (epoch), it can be configured to a different value.

I didn't want to introduce a behaviour, which wasn't possible before.